src/ring_theory/dedekind_domain/dvr.lean

```diff
@@ -52,3 +61,100 @@ structure is_dedekind_domain_dvr : Prop :=
 (is_noetherian_ring : is_noetherian_ring A)
 (is_dvr_at_nonzero_prime : ∀ P ≠ (⊥ : ideal A), P.is_prime →
   discrete_valuation_ring (localization.at_prime P))
+
+lemma is_localization.bot_lt_comap_prime {R : Type*} (Rₘ : Type*) [comm_ring R] [comm_ring Rₘ]
+  [is_domain R] [algebra R Rₘ] (M : submonoid R) [is_localization M Rₘ] (hM : M ≤ R⁰)
+  (p : ideal Rₘ) [hpp : p.is_prime] (hp0 : p ≠ ⊥) :
+  ⊥ < ideal.comap (algebra_map R Rₘ) p :=
+begin
+  haveI : is_domain Rₘ := is_localization.is_domain_of_le_non_zero_divisors _ hM,
+  convert (is_localization.order_iso_of_prime M Rₘ).lt_iff_lt.mpr
+    (show (⟨⊥, ideal.bot_prime⟩ : {p : ideal Rₘ // p.is_prime}) < ⟨p, hpp⟩, from hp0.bot_lt),
+  exact (ideal.comap_bot_of_injective (algebra_map R Rₘ) (is_localization.injective _ hM)).symm,
+end
+
+lemma ring.dimension_le_one.localization {R : Type*} (Rₘ : Type*) [comm_ring R] [is_domain R]
```

Comment:
Does a similar proof generalize to show that the dimension of the localization is less or equal to the dimension of the ring?

Reply:
It should be exactly equal I believe, due to `is_localization.order_iso_of_prime`. I don't know the best way to express the Krull dimension property though: something like `∀ (Ps : list (ideal P)), (∀ P ∈ Ps, P.is_prime) → Ps.pairwise (<) → Ps.length ≤ n`?

Reply:
I thought we had `krull_dimension` and `dimension_le_one` was defined using it. Since it's the explicit definition terms of prime/maximal ideal let's just leave a comment saying the same proof works in general. At some point someone will define the dimension in full generality.

Reply:
```suggestion
/-- Localizing a domain of Krull dimension `≤ 1` gives another ring of Krull dimension `≤ 1`.

Note that the same proof can/should be generalized to preserving any Krull dimension,
once we have a suitable definition.
-/
lemma ring.dimension_le_one.localization {R : Type*} (Rₘ : Type*) [comm_ring R] [is_domain R]
```